lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs

```diff
@@ -0,0 +1,113 @@
+{-# LANGUAGE LambdaCase #-}
+module Cardano.Wallet.Jormungandr.NetworkSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Launcher
+    ( Command (..), StdStream (..), launch )
+import Cardano.Wallet.Jormungandr.Api
+    ( BlockId (..) )
+import Cardano.Wallet.Jormungandr.Network
+    ( BaseUrl (..)
+    , Scheme (Http)
+    , defaultManagerSettings
+    , getBlock
+    , getTipId
+    , mkJormungandr
+    , newManager
+    , postSignedTx
+    )
+import Cardano.Wallet.Network
+    ( ErrNetworkTip (..), ErrNetworkUnreachable (..), ErrPostTx (..) )
+import Cardano.Wallet.Primitive.Types
+    ( Block (..), Hash (..), Tx (..) )
+import Control.Concurrent
+    ( threadDelay )
+import Control.Concurrent.Async
+    ( async, cancel )
+import Control.Monad.Trans.Except
+    ( runExceptT )
+import Test.Hspec
+    ( Spec
+    , afterAll
+    , beforeAll
+    , describe
+    , it
+    , shouldBe
+    , shouldReturn
+    , shouldSatisfy
+    , xit
+    )
+
+import qualified Data.ByteString as BS
+
+spec :: Spec
+spec = do
+    describe "Happy paths"
+        $ beforeAll startNode $ afterAll closeNode $ do
+        it "get tip (should be genesis)" $ \(_, client) -> do
+            (BlockId (Hash tipHash)) <- run $ getTipId client
+            tipHash `shouldBe` genesisHash
+            (BS.length tipHash) `shouldBe` 32
+
+        it "get block (genesis)" $ \(_, client) -> do
+            b <- run $ getBlock client (BlockId $ Hash genesisHash)
+            length (transactions b) `shouldBe` 1
+
+    describe "Error paths" $ beforeAll newClient $ do
+        it "gets a 'ErrNetworkUnreachable' if jormungandr isn't up (1)"
```

Comment:
For `http-bridge` we test this on the `NetworkLayer`. I used the `Jormungandr`-client here as `NetworkLayer` isn't implemented yet.